crates/cast/src/lib.rs

```diff
@@ -163,6 +163,20 @@ pub fn decode_chain_id(chain_id: FieldElement) -> String {
     String::from_utf8(non_zero_bytes).unwrap_or_default()
 }
 
+pub async fn get_nonce(
+    provider: &JsonRpcClient<HttpTransport>,
+    block_id: &str,
+    address: FieldElement,
+) -> Result<FieldElement> {
+    Ok(provider
+        .get_nonce(
+            get_block_id(block_id).expect("Could not obtain block id"),
+            address,
+        )
+        .await
+        .expect("Could not get current nonce"))
```

Comment:
Nitpick, but this function is not necessarily for fetching a "current" nonce, but for any given `block_id`, right?

Reply:
For now only for block tags (pending, latest). Removed word "current" in newest commit.

Reply:
no use for it currently, that's the reason for limitation (and it is limited on scripts side, so actually make sense to remove "current" from this function)